docs/ske/10-integrations/20-port.mdx

```diff
@@ -0,0 +1,347 @@
+          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
+          operation: PATCH_RUN
+          runId: ${{inputs.runId}} 
+          icon: GithubActions
+          logMessage: |
+            Successfully committed request to main ✅
+      - uses: port-labs/port-github-action@v1
+        if: failure()
+        with:
+          clientId: ${{ secrets.PORT_CLIENT_ID }}
+          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
+          operation: PATCH_RUN
+          runId: ${{inputs.runId}} 
+          icon: GithubActions
+          logMessage: | 
+            Failed to commit request to main ❌
+            Resource may have been created outside of Port 🕵
+            Check the job run logs in Github to learn more 📜:
+              ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+```
+
+</details>
+
+
+## Configure GitOps for requests from Port Actions {#gitops}
+
+So far you have configured Port to use GitHub Actions as a backend for the Port
+self-serve actions and automations. You have then configured the GitHub Action to
+commit user requests to the defined repository.
+
+Now you will need to configure a GitOps agent to pull the repository documents into the
+cluster where Kratix runs. This will enable Kratix to respond to resources that are
+created via Port.
+
+Please refer to the guide documentation to use
+[ArgoCD](https://docs.kratix.io/main/guides/installing-argocd),
+[Flux](https://docs.kratix.io/main/guides/installing-fluxcd), or any
+[other agent](https://docs.kratix.io/main/guides/custom-gitops-agent) you prefer.
+
+Part of the configuration is making sure that the GitOps agent is reading from the
+repository that Port is writing to. An example set up when using FluxCD and the defaults
+from above can be found below.
+
+<details>
+  <summary>Example Flux Resources for Port-Syncer repository</summary>
+
+```
+---
+apiVersion: source.toolkit.fluxcd.io/v1
+kind: GitRepository
+metadata:
+  name: kratix-port-resources
+  namespace: flux-system
+spec:
+  interval: 5s
+  url: https://github.com/${PORT_GITHUB_ORG}/${PORT_GITHUB_REPO}
+  ref:
+    branch: main
+  secretRef:
+    name: github-creds
+---
+apiVersion: kustomize.toolkit.fluxcd.io/v1
+kind: Kustomization
+metadata:
+  name: kratix-port-resources
+  namespace: flux-system
+spec:
+  interval: 3s
+  sourceRef:
+    kind: GitRepository
+    name: kratix-port-resources
+  path: "kratix/platform/resources/"
+  prune: true
+```
+
+</details>
+
+## Expose a Promise in Port {#promise-in-port}
+
+While many platform APIs benefit from being available, you may not want _all_ platform
+APIs in the Port UI. For this reason, Kratix uses an opt-in methodology.
+
+To expose a Promise via the Port UI, make sure to add a label
+`kratix.io/get-port: "true"` in the Kratix Promise API. For example, to expose the
+[marketplace namespace promise](https://github.com/syntasso/kratix-marketplace/tree/main/namespace),
+it would look like this:
```

Comment:
I think its fine to ship now, but in future would be interesting if we can change it so the Promise needs to be labelled, not the CRD within it. I feel like it makes it a bit cleaner. Not sure how well the Kratix controller would play with some other tool adding a label to the CRD it manages

Reply:
100, this is part of the architecture review I was talking about earlier 💯